prow/rewriteConfig.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"regexp"
+	"strings"
+)
+
+// Count the number of spaces before a valid character in a string.
+func countLeadingSpace(line string) int {
+	i := 0
+	for _, runeValue := range line {
+		if runeValue == ' ' {
+			i++
+		} else {
+			break
+		}
+	}
+	return i
+}
+
+// Add element to the slice at the indexed position and move all the strings from
+// the index one position to the right.
+func appendAtIndex(original string, element string, index int) string {
+	prev := original[:index]
+	apres := original[index:]
+	return prev + element + apres
+}
+
+// Find index of master section in each branch split for the new branch and
+// get content from master section with mandatory "merges-blocked-needs-admin" part.
+func getContentOfNewBranch(branchContent string, newBranch string) (int, string) {
+	branchLines := strings.Split(branchContent, "\n")
+	if len(branchLines) < 2 {
+		return -1, ""
+	}
+
+	firstBranchLine := branchLines[1]
+	spacesForBranchNum := countLeadingSpace(firstBranchLine)
+
+	spacesForBranch := strings.Repeat(" ", spacesForBranchNum)
+	masterLine := spacesForBranch + "master:\n"
+	masterStart := strings.Index(branchContent, masterLine)
+
+	nonMasterContent := branchContent[masterStart+1:]
+
+	generalBranchRe := regexp.MustCompile("\n" + spacesForBranch + "\\S+:\n")
+	otherBranches := generalBranchRe.FindStringSubmatchIndex(nonMasterContent)
+
+	var masterContent string
+	if otherBranches == nil {
+		masterContent = branchContent[masterStart:]
+	} else {
+		masterContent = branchContent[masterStart:otherBranches[0]]
+	}
+
+	masterContent = strings.Join(strings.Split(masterContent, "\n")[1:], "\n")
+	contextRe := regexp.MustCompile("\\s+contexts:\n")
+	contextInd := contextRe.FindStringSubmatchIndex(masterContent)
+	if contextInd == nil {
+		masterContentSplit := strings.Split(masterContent, "\n")
+		masterContentLine := masterContentSplit[0]
+		spacesInMaster := countLeadingSpace(masterContentLine)
+
+		tab := spacesInMaster - spacesForBranchNum
+		statusCheckString := strings.Repeat(" ", spacesInMaster) + "required_status_checks:\n"
+		contextsString := strings.Repeat(" ", spacesInMaster+tab) + "contexts:\n"
+
+		if strings.Compare(masterContentSplit[len(masterContentSplit)-1], "") != 0 {
```

Comment:
why do you need this check?

Reply:
I cannot say for sure whether or not the previous line ends with a "\n" so if I just add the lines directly it might end up trailing after the previous line without having it's own line.